src/containers/SecretsModal/SecretsModal.js

```diff
@@ -47,6 +47,32 @@ function validateInputs(value, id) {
   return true;
 }
 
+// Used to determine if an annotation has been added or removed and therefore if the modal should scroll down
+function handleScroll(state, prevState) {
+  let shouldScroll = true;
```

Comment:
do we need all of these checks or can we simplify and only check the annotations?

Reply:
Unfortunately this is the only way Adam and I found of checking it.. I realise it's inefficient but there is no way of checking the annotations against the previous state as they get updated much before this point is reached.